fabric-dimensions-v1/src/main/java/net/fabricmc/fabric/api/dimension/EntityPlacer.java

```diff
@@ -0,0 +1,32 @@
+package net.fabricmc.fabric.api.dimension;
+
+import net.minecraft.block.pattern.BlockPattern;
+import net.minecraft.entity.Entity;
+import net.minecraft.server.world.ServerWorld;
+import net.minecraft.util.math.Direction;
+
+/**
+ * Responsible for placing an Entity once they have entered a dimension.
+ * Stored by a FabricDimensionType, and used in Entity::changeDimension.
+ *
+ * @see FabricDimensions
+ * @see FabricDimensionType
+ */
+@FunctionalInterface
+public interface EntityPlacer {
+
+    /**
+     * Handles the placement of an entity going to a dimension.
+     *
+     * <p> This method may have side effects such as the creation of a portal in the target dimension,
+     * or the creation of a chunk loading ticket.
+     *
+     * @param portalDir the direction the portal is facing, meaningless if no portal was used
+     * @param horizontalOffset the horizontal offset of the entity relative to the front top left corner of the portal, meaningless if no portal was used
+     * @param verticalOffset  the vertical offset of the entity relative to the front top left corner of the portal, meaningless if no portal was used
+     * @return a teleportation target, or {@code null} to fall back to further handling
```

Comment:
"further handling"?

Reply:
Refer to the documentation in [FabricDimensions#teleport](https://github.com/FabricMC/fabric/pull/319/files#diff-6e20c61082cebd768d952720aeb2ac4eR20). That method supports `null` returns, `FabricDimensionType#defaultPlacement` does not. I guess this should be added to the javadoc.

Reply:
I'd use something like "..to use the dimension's default behavior, see {@link.."